src/tools/export-plan/pdf-schedule-generator.ts

```diff
@@ -0,0 +1,506 @@
+import summerEmojiURL from '@/assets/images/pdf-gen/summer.png';
+import winterEmojiURL from '@/assets/images/pdf-gen/winter.png';
+import APIBEmojiURL from '@/assets/images/pdf-gen/apib.png';
+import { lightPlaceholderGray, borderGray } from '@/assets/constants/scss-variables';
+import { pdfColors } from '@/assets/constants/colors';
+import { bubbleColorMap, loadImage, getCourseRowsWithForcedReqs } from './utilities';
+import { SemesterRows, ReqInfo } from './types';
+
+import {
+  getCollegeFullName,
+  getMajorFullName,
+  getMinorFullName,
+  getGradFullName,
+} from '../../utilities';
+import store from '../../store';
+import { addFonts } from './add-fonts';
+
+const rowHeight = 18;
+const tableWidth = 516;
+// vertical space between tables for two semesters
+const tableGap = 20;
+// starting y co-ordinate of the first table
+const firstPageFirstTableY = 90;
+const secondPageFirstTableY = 170;
+
+const rowFontSize = 10.5;
+const headerFontSize = 10.5;
+const lineSpacing = 1.5;
+const headerLinesGap = 7;
+
+const tableHeader = [['Course', 'Credits', 'Requirements Fulfilled']];
+
+// max number of characters that can fit into a line for the major, minor or grad program field
+const programLineCharLimit = 45;
+
+/**
+ * Generates a PDF of the schedule for the given classes.
+ *
+ * @param classes a mapping of a requirement to the autogenerated course that fulfills it. Currently has an edge case in that it can only handle one course per requirement.
+ * @param calendarDataString the base64 encoded string of the calendar image
+ * @param year the year of the generated semester schedule
+ * @param season the season of the generated semester schedule
+ */
+const generateSchedulePDF = async (
+  classes: Map<ReqInfo, FirestoreSemesterCourse>,
+  calendarDataString: string,
+  year: number,
+  season: FirestoreSemesterSeason
+): Promise<void> => {
+  const doc = new JsPDF({ unit: 'pt', format: 'letter' });
+
+  addFonts(doc);
+  doc.setFont('ProximaNova-Regular', 'normal');
+
+  const tableX = (doc.internal.pageSize.width - tableWidth) / 2;
+  let programY = 110;
+
+  // render PDF header for page 1
+  await renderFirstHeader(doc);
+
+  // Rendering table now
+  const tableHeight = await renderBaseTable(doc, classes, year, season, programY, tableX);
+
+  // Calendar time
+  const imgProps = doc.getImageProperties(calendarDataString);
+
+  // Sh
+
+  doc.addImage(
+    calendarDataString,
+    'PNG',
+    48,
+    tableHeight,
+    tableWidth,
+    imgProps.height * (tableWidth / imgProps.width)
+  );
+
+  renderFooter(doc);
+
+  // create new page
+  doc.addPage();
+
+  doc.setPage(2);
+
+  // rendering PDF header for page 2
+  programY = await renderSecondHeader(doc);
+
+  // Rendering tables now
+  await renderBaseTable(doc, classes, year, season, programY, tableX, 'right');
+
+  // rendering PDF footer
+  renderFooter(doc);
+
+  const pdfName = `${store.state.userName.firstName}_CoursePlan`;
```

Comment:
could you call this something else like "Generated_Schedule" to differentiate it from the other PDF download?

Reply:
Shoot sorry I missed this comment somehow, fixed on another branch.